app/main.py

```diff
@@ -86,129 +89,20 @@
 }
 
 
-def battle(knightsConfig):
+def battle(knights_config: dict) -> dict:
     # BATTLE PREPARATIONS:
+    knights_obj_list = Knight.create_knight_objects(knights_config)
+    prepared_knights = knights_preparation(knights_obj_list)
```

Comment:
You can unpack knights and put them in variables and refactor `start_battle` so that it accepts knights that should battle only